### PR TITLE
add repo-sync orb

### PR DESCRIPTION
**Pull request by gabrielittner**
Commands are copied from here: https://github.com/freeletics/fl-api-contracts/blob/master/.circleci/config.yml

The ci config is copied from one of the other orbs

### CODE REVIEW

File: repo-sync/commands/publish_to_repo.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - run:
+      name: Install hub command (Linux)
+      command: |
+        cd ../
+        wget -O hub.tgz https://github.com/github/hub/releases/download/v2.13.0/hub-linux-amd64-2.13.0.tgz
+        tar zxv -f hub.tgz --strip-components=2 hub-linux-amd64-2.13.0/bin/hub
+  - run:
+      name: Creating new pull request
+      command: |
+        ./.circleci/write-pr-file.sh
```
Comment (mathaeus) [inline repo-sync/commands/publish_to_repo.yml:39]: That file is missing from what I can see.
Or which `circleci` folder does this refer to? Not sure by just looking at the code like this.

File: repo-sync/commands/publish_to_repo.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      command: |
+        mkdir ../targetRepository
+        git clone --depth 5 git@github.com:freeletics/<<parameters.target_repository>>.git ../targetRepository
+        pushd ../targetRepository
+        git checkout -b <<parameters.pr_label>>/$CIRCLE_BUILD_NUM <<parameters.target_branch>>
+        popd
+  - steps: <<parameters.update_steps>>
+  - abort_if_no_changes
+  - run:
+      name: Install hub command (Linux)
```
Comment (mathaeus) [inline repo-sync/commands/publish_to_repo.yml:31]: Does the Linux command work if the orb is executed in a command that is part of a job of a Mac executor? 🤪

Comment (gabrielittner) [reply repo-sync/commands/publish_to_repo.yml]: we had an extra macOS download in the tracking repo, I've combined it into a single step that automatically does the right thing based on platform: 0698979

File: repo-sync/commands/publish_to_repo.yml (hunk 1 above)
Comment (gabrielittner) [reply repo-sync/commands/publish_to_repo.yml]: moved over 61abe17

**Review by mathaeus — APPROVED** [review]
Let's give it a try 🙇

**Comment by gabrielittner** [comment]
Had to fix the echo. Successful Android run https://github.com/freeletics/fl-application-android/pull/13541 (only used Android because that's the only platform that always publishes, but for api contracts at least there is not much platform dependent logic so it should be fine)

**Pull request merged**